File: app/src/main/java/org/deiverbum/app/data/entity/TodayEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.TODAY_TABLE;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */

@Entity(tableName = TODAY_TABLE,
        indices = {@Index(value = {"todayDate","timeID","liturgyFK","previousFK","massReadingFK","invitatoryFK","saintFK","oHymnFK","oPsalmodyFK","oVerseFK","oBiblicalFK","oPatristicFK","oPrayerFK","oTeDeum","lHymnFK","lPsalmodyFK","lBiblicalFK","lBenedictusFK","lIntercessionsFK","lPrayerFK","tHymnFK","tPsalmodyFK","tBiblicalFK","tPrayerFK","sHymnFK","sPsalmodyFK","sBiblicalFK","sPrayerFK","nHymnFK","nPsalmodyFK","nBiblicalFK","nPrayerFK","vHymnFK","vPsalmodyFK","vBiblicalFK","vMagnificatFK","vIntercessionsFK","vPrayerFK"},unique = true)},

        foreignKeys =
        {
                @ForeignKey(
                        entity = LiturgyEntity.class,
                        parentColumns = "liturgyID",
                        childColumns = "liturgyFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),

                @ForeignKey(
                        entity = LiturgyEntity.class,
                        parentColumns = "liturgyID",
                        childColumns = "previousFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = MassReadingJoinEntity.class,
                        parentColumns = "liturgyFK",
                        childColumns = "massReadingFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHInvitatoryJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "invitatoryFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHHymnJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "oHymnFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPsalmodyJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "oPsalmodyFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHOficceVerseJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "oVerseFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHOfficeBiblicalJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "oBiblicalFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHOfficePatristicJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "oPatristicFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = SaintEntity.class,
                        parentColumns = "saintID",
                        childColumns = "saintFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPrayerEntity.class,
                        parentColumns = "groupID",
                        childColumns = "oPrayerFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHHymnJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "lHymnFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPsalmodyJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "lPsalmodyFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHGospelCanticleJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "lBenedictusFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHReadingShortJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "lBiblicalFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHIntercessionsJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "lIntercessionsFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPrayerEntity.class,
                        parentColumns = "groupID",
                        childColumns = "lPrayerFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHHymnJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "tHymnFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPsalmodyJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "tPsalmodyFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHReadingShortJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "tBiblicalFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPrayerEntity.class,
                        parentColumns = "groupID",
                        childColumns = "tPrayerFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHHymnJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "sHymnFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPsalmodyJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "sPsalmodyFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHReadingShortJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "sBiblicalFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPrayerEntity.class,
                        parentColumns = "groupID",
                        childColumns = "sPrayerFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHHymnJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "nHymnFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPsalmodyJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "nPsalmodyFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHReadingShortJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "nBiblicalFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPrayerEntity.class,
                        parentColumns = "groupID",
                        childColumns = "nPrayerFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHHymnJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "vHymnFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPsalmodyJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "vPsalmodyFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHGospelCanticleJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "vMagnificatFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHReadingShortJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "vBiblicalFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHIntercessionsJoinEntity.class,
                        parentColumns = "groupID",
                        childColumns = "vIntercessionsFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT),
                @ForeignKey(
                        entity = LHPrayerEntity.class,
                        parentColumns = "groupID",
                        childColumns = "vPrayerFK",
                        onDelete = ForeignKey.RESTRICT,
                        onUpdate = ForeignKey.RESTRICT)
        })
public class TodayEntity {
    @NonNull
    @PrimaryKey
    @ColumnInfo(name = "todayDate")
    public Integer hoy=0;

    @NonNull
    @ColumnInfo(name = "timeID", defaultValue = "1")
    public Integer tiempoId=1;

    @NonNull
    @ColumnInfo(name = "weekDay", defaultValue = "1")
    public Integer weekDay=1;

    @NonNull
    @ColumnInfo(name = "liturgyFK", defaultValue = "1")
    public Integer liturgyFK=1;

    @NonNull
    @ColumnInfo(name = "previousFK", defaultValue = "1")
    public Integer previoId=1;

    @NonNull
    @ColumnInfo(name = "hasSaint", defaultValue= "0")
    public Integer hasSaint=0;

    @NonNull
    @ColumnInfo(name = "massReadingFK")
    public Integer mLecturasFK=0;
    
    @NonNull
    @ColumnInfo(name = "invitatoryFK")
    public Integer invitatorioFK=0;

    @NonNull
    @ColumnInfo(name = "saintFK")
    public Integer santoFK=1;

    @NonNull
    @ColumnInfo(name = "oHymnFK")
    public Integer oHimnoFK=0;

    @NonNull
    @ColumnInfo(name = "oPsalmodyFK")
    public Integer oSalmodiaFK=0;
    
    @NonNull
    @ColumnInfo(name = "oVerseFK")
    public Integer oVersoFK=0;

    @NonNull
    @ColumnInfo(name = "oBiblicalFK")
    public Integer oBiblicaFK=0;

    @NonNull
    @ColumnInfo(name = "oPatristicFK")
    public Integer oPatristicaFK=0;

    @NonNull
    @ColumnInfo(name = "oTeDeum", defaultValue= "0")
    public Integer oTeDeum=0;

    @NonNull
    @ColumnInfo(name = "oPrayerFK")
    public Integer oOracionFK=0;

    @NonNull
    @ColumnInfo(name = "lHymnFK")
    public Integer lHimnoFK=0;

    @NonNull
    @ColumnInfo(name = "lPsalmodyFK")
    public Integer lSalmodiaFK=0;

    @NonNull
    @ColumnInfo(name = "lBiblicalFK")
    public Integer lBiblicaFK=0;

    @NonNull
    @ColumnInfo(name = "lBenedictusFK")
    public Integer lBenedictusFK=0;

    @NonNull
    @ColumnInfo(name = "lIntercessionsFK")
    public Integer lPrecesFK=0;

    @NonNull
    @ColumnInfo(name = "lPrayerFK")
    public Integer lOracionFK=0;


    @NonNull
    @ColumnInfo(name = "tHymnFK")
    public Integer tHimnoFK=0;

    @NonNull
    @ColumnInfo(name = "tPsalmodyFK")
    public Integer tSalmodiaFK=0;

    @NonNull
    @ColumnInfo(name = "tBiblicalFK")
    public Integer tBiblicaFK=0;

    @NonNull
    @ColumnInfo(name = "tPrayerFK")
    public Integer tOracionFK=0;

    @NonNull
    @ColumnInfo(name = "sHymnFK")
    public Integer sHimnoFK=0;

    @NonNull
    @ColumnInfo(name = "sPsalmodyFK")
    public Integer sSalmodiaFK=0;

    @NonNull
    @ColumnInfo(name = "sBiblicalFK")
    public Integer sBiblicaFK=0;

    @NonNull
    @ColumnInfo(name = "sPrayerFK")
    public Integer sOracionFK=0;

    @NonNull
    @ColumnInfo(name = "nHymnFK")
    public Integer nHimnoFK=0;

    @NonNull
    @ColumnInfo(name = "nPsalmodyFK")
    public Integer nSalmodiaFK=0;

    @NonNull
    @ColumnInfo(name = "nBiblicalFK")
    public Integer nBiblicaFK=0;

    @NonNull
    @ColumnInfo(name = "nPrayerFK")
    public Integer nOracionFK=0;


    @NonNull
    @ColumnInfo(name = "vHymnFK")
    public Integer vHimnoFK=0;

    @NonNull
    @ColumnInfo(name = "vPsalmodyFK")
    public Integer vSalmodiaFK=0;

    @NonNull
    @ColumnInfo(name = "vBiblicalFK")
    public Integer vBiblicaFK=0;

    @NonNull
    @ColumnInfo(name = "vMagnificatFK")
    public Integer vMagnificatFK=0;

    @NonNull
    @ColumnInfo(name = "vIntercessionsFK")
    public Integer vPrecesFK=0;

    @NonNull
    @ColumnInfo(name = "vPrayerFK")
    public Integer vOracionFK=0;

    @NonNull
    public Integer getHoy() {
        return hoy;
    }
    public void setHoy(@NonNull Integer hoy) {
        this.hoy = hoy;
    }

    @NonNull
    public Integer getPrevioId() {
        return previoId;
    }

    public void setPrevioId(@NonNull Integer previoId) {
        this.previoId = previoId;
    }

    @NonNull
    public Integer getTiempoId() {
        return tiempoId;
    }

    @SuppressWarnings("unused")
    public boolean getTeDeum() {
        return oTeDeum == 1;
    }
}

File: app/src/main/java/org/deiverbum/app/data/entity/TodayEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.*
import androidx.room.ForeignKey.Companion.CASCADE
import org.deiverbum.app.util.Constants

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(
    tableName = Constants.TODAY_TABLE,
    indices = [Index(
        value = ["todayDate", "timeID", "liturgyFK", "previousFK", "massReadingFK", "invitatoryFK", "saintFK", "oHymnFK", "oPsalmodyFK", "oVerseFK", "oBiblicalFK", "oPatristicFK", "oPrayerFK", "oTeDeum", "lHymnFK", "lPsalmodyFK", "lBiblicalFK", "lBenedictusFK", "lIntercessionsFK", "lPrayerFK", "tHymnFK", "tPsalmodyFK", "tBiblicalFK", "tPrayerFK", "sHymnFK", "sPsalmodyFK", "sBiblicalFK", "sPrayerFK", "nHymnFK", "nPsalmodyFK", "nBiblicalFK", "nPrayerFK", "vHymnFK", "vPsalmodyFK", "vBiblicalFK", "vMagnificatFK", "vIntercessionsFK", "vPrayerFK", "nightPrayerFK"],
        unique = true
    )],
    foreignKeys = [ForeignKey(
        entity = LiturgyEntity::class,
        parentColumns = arrayOf("liturgyID"),
        childColumns = arrayOf("liturgyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LiturgyEntity::class,
        parentColumns = arrayOf("liturgyID"),
        childColumns = arrayOf("previousFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = MassReadingJoinEntity::class,
        parentColumns = arrayOf("liturgyFK"),
        childColumns = arrayOf("massReadingFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHInvitatoryJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("invitatoryFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHHymnJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("oHymnFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPsalmodyJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("oPsalmodyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHOfficeVerseJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("oVerseFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHOfficeBiblicalJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("oBiblicalFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHOfficePatristicJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("oPatristicFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = SaintEntity::class,
        parentColumns = arrayOf("saintID"),
        childColumns = arrayOf("saintFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPrayerEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("oPrayerFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHHymnJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("lHymnFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPsalmodyJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("lPsalmodyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHGospelCanticleEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("lBenedictusFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHReadingShortJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("lBiblicalFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHIntercessionsJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("lIntercessionsFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPrayerEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("lPrayerFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHHymnJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("tHymnFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPsalmodyJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("tPsalmodyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHReadingShortJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("tBiblicalFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPrayerEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("tPrayerFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHHymnJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("sHymnFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPsalmodyJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("sPsalmodyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHReadingShortJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("sBiblicalFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPrayerEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("sPrayerFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHHymnJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("nHymnFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPsalmodyJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("nPsalmodyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHReadingShortJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("nBiblicalFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPrayerEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("nPrayerFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHHymnJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("vHymnFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPsalmodyJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("vPsalmodyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHGospelCanticleEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("vMagnificatFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHReadingShortJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("vBiblicalFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHIntercessionsJoinEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("vIntercessionsFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHPrayerEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("vPrayerFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHNightPrayerEntity::class,
        parentColumns = arrayOf("groupID"),
        childColumns = arrayOf("nightPrayerFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    )
    ]
)
class TodayEntity {
    @JvmField
    @PrimaryKey
    @ColumnInfo(name = "todayDate")
    var hoy = 0

    @JvmField
    @ColumnInfo(name = "timeID", index = true)
    var tiempoId = 1

    @JvmField
    @ColumnInfo(name = "weekDay", defaultValue = "1", index = true)
    var weekDay = 1

    @JvmField
    @ColumnInfo(name = "liturgyFK", index = true)
    var liturgyFK = 1

    @JvmField
    @ColumnInfo(name = "previousFK", index = true)
    var previoId = 1

    @JvmField
    @ColumnInfo(name = "hasSaint", defaultValue = "0")
    var hasSaint = 0

    @JvmField
    @ColumnInfo(name = "massReadingFK", index = true)
    var mLecturasFK = 0

    @JvmField
    @ColumnInfo(name = "invitatoryFK", index = true)
    var invitatorioFK = 0

    @JvmField
    @ColumnInfo(name = "saintFK", index = true)
    var santoFK = 1

    @JvmField
    @ColumnInfo(name = "oHymnFK", index = true)
    var oHimnoFK = 0

    @JvmField
    @ColumnInfo(name = "oPsalmodyFK", index = true)
    var oSalmodiaFK = 0

    @JvmField
    @ColumnInfo(name = "oVerseFK", index = true)
    var oVersoFK = 0

    @JvmField
    @ColumnInfo(name = "oBiblicalFK", index = true)
    var oBiblicaFK = 0

    @JvmField
    @ColumnInfo(name = "oPatristicFK", index = true)
    var oPatristicaFK = 0

    @JvmField
    @ColumnInfo(name = "oTeDeum", index = true)
    var oTeDeum = 0

    @JvmField
    @ColumnInfo(name = "oPrayerFK", index = true)
    var oOracionFK = 0

    @JvmField
    @ColumnInfo(name = "lHymnFK", index = true)
    var lHimnoFK = 0

    @JvmField
    @ColumnInfo(name = "lPsalmodyFK", index = true)
    var lSalmodiaFK = 0

    @JvmField
    @ColumnInfo(name = "lBiblicalFK", index = true)
    var lBiblicaFK = 0

    @JvmField
    @ColumnInfo(name = "lBenedictusFK", index = true)
    var lBenedictusFK = 0

    @JvmField
    @ColumnInfo(name = "lIntercessionsFK", index = true)
    var lPrecesFK = 0

    @JvmField
    @ColumnInfo(name = "lPrayerFK", index = true)
    var lOracionFK = 0

    @JvmField
    @ColumnInfo(name = "tHymnFK", index = true)
    var tHimnoFK = 0

    @JvmField
    @ColumnInfo(name = "tPsalmodyFK", index = true)
    var tSalmodiaFK = 0

    @JvmField
    @ColumnInfo(name = "tBiblicalFK", index = true)
    var tBiblicaFK = 0

    @JvmField
    @ColumnInfo(name = "tPrayerFK", index = true)
    var tOracionFK = 0

    @JvmField
    @ColumnInfo(name = "sHymnFK", index = true)
    var sHimnoFK = 0

    @JvmField
    @ColumnInfo(name = "sPsalmodyFK", index = true)
    var sSalmodiaFK = 0

    @JvmField
    @ColumnInfo(name = "sBiblicalFK", index = true)
    var sBiblicaFK = 0

    @JvmField
    @ColumnInfo(name = "sPrayerFK", index = true)
    var sOracionFK = 0

    @JvmField
    @ColumnInfo(name = "nHymnFK", index = true)
    var nHimnoFK = 0

    @JvmField
    @ColumnInfo(name = "nPsalmodyFK", index = true)
    var nSalmodiaFK = 0

    @JvmField
    @ColumnInfo(name = "nBiblicalFK", index = true)
    var nBiblicaFK = 0

    @JvmField
    @ColumnInfo(name = "nPrayerFK", index = true)
    var nOracionFK = 0

    @JvmField
    @ColumnInfo(name = "vHymnFK", index = true)
    var vHimnoFK = 0

    @JvmField
    @ColumnInfo(name = "vPsalmodyFK", index = true)
    var vSalmodiaFK = 0

    @JvmField
    @ColumnInfo(name = "vBiblicalFK", index = true)
    var vBiblicaFK = 0

    @JvmField
    @ColumnInfo(name = "vMagnificatFK", index = true)
    var vMagnificatFK = 0

    @JvmField
    @ColumnInfo(name = "vIntercessionsFK", index = true)
    var vPrecesFK = 0

    @JvmField
    @ColumnInfo(name = "vPrayerFK", index = true)
    var vOracionFK = 0

    @JvmField
    @ColumnInfo(name = "nightPrayerFK", defaultValue = "71", index = true)
    var nightPrayerFK = 71

    /*val teDeum: Boolean
        get() = oTeDeum == 1*/
}